src/analysis/complex/unit_disc/basic.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2022 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import analysis.complex.circle
+import analysis.normed_space.ball_action
+import group_theory.subsemigroup.membership
+
+/-!
+# Poincaré disc
+
+In this file we define `complex.unit_disc` to be the unit disc in the complex plane. We also
+introduce some basic operations on this disc.
+-/
+
+open set function metric
+open_locale big_operators
+noncomputable theory
+
+local notation `conj'` := star_ring_end ℂ
+
+namespace complex
+
+/-- Complex unit disc. -/
+@[derive [comm_semigroup, has_distrib_neg, λ α, has_coe α ℂ, topological_space]]
+def unit_disc : Type := ball (0 : ℂ) 1
+localized "notation `𝔻` := complex.unit_disc" in unit_disc
+
+namespace unit_disc
+
+lemma coe_injective : injective (coe : 𝔻 → ℂ) := subtype.coe_injective
+
+lemma abs_lt_one (z : 𝔻) : abs (z : ℂ) < 1 := mem_ball_zero_iff.1 z.2
+
+lemma abs_ne_one (z : 𝔻) : abs (z : ℂ) ≠ 1 := z.abs_lt_one.ne
+
+lemma norm_sq_lt_one (z : 𝔻) : norm_sq z < 1 :=
+@one_pow ℝ _ 2 ▸ (real.sqrt_lt' one_pos).1 z.abs_lt_one
+
+lemma coe_ne_one (z : 𝔻) : (z : ℂ) ≠ 1 :=
+ne_of_apply_ne abs $ (map_one abs).symm ▸ z.abs_ne_one
+
+lemma coe_ne_neg_one (z : 𝔻) : (z : ℂ) ≠ -1 :=
+ne_of_apply_ne abs $ by { rw [abs.map_neg, map_one], exact z.abs_ne_one }
+
+lemma one_add_coe_ne_zero (z : 𝔻) : (1 + z : ℂ) ≠ 0 :=
+mt neg_eq_iff_add_eq_zero.2 z.coe_ne_neg_one.symm
+
+@[simp, norm_cast] lemma coe_mul (z w : 𝔻) : ↑(z * w) = (z * w : ℂ) := rfl
+
+/-- A constructor that assumes `abs z < 1` instead of `dist z 0 < 1` returns an element of `𝔻`
+instead of `↥metric.ball (0 : ℂ) 1`. -/
```

Comment:
```suggestion
/-- A constructor that assumes `abs z < 1` instead of `dist z 0 < 1` and returns an element 
of `𝔻` instead of `↥metric.ball (0 : ℂ) 1`. -/
```